pkg/policy/api/rule.go

```diff
@@ -0,0 +1,228 @@
+	// If omitted, this rule does not apply at ingress.
+	//
+	// +optional
+	Ingress []IngressRule `json:"ingress,omitempty"`
+
+	// Egress is a list of EgressRule which are enforced at egress.
+	// If omitted, this rule does not apply at ingress.
+	//
+	// +optional
+	Egress []EgressRule `json:"egress,omitempty"`
+
+	// Labels is a list of optional strings which can be used to
+	// re-identify the rule or to store metadata. It is possible to lookup
+	// or delete strings based on labels. Labels are not required to be
+	// unique, multiple rules can have overlapping or identical labels.
+	//
+	// +optional
+	Labels labels.LabelArray `json:"group,omitempty"`
+
+	// Description is a free form string, it can be used by the creator of
+	// the rule to store human readable explanation of the purpose of this
+	// rule. Rules cannot be identified by comment.
+	//
+	// +optional
+	Description string `json:"description,omitempty"`
+}
+
+// IngressRule contains all rule types which can be applied at ingress,
+// i.e. network traffic that originates outside of the endpoint and
+// is entering the endpoint selected by the endpointSelector.
+//
+// - All members of this structure are optional.
+// - All rule types are considered independently
+type IngressRule struct {
+	// FromEndpoints is a list of endpoints identified by an
+	// EndpointSelector which are allowed to communicate with the endpoint
+	// subject to the rule.
+	//
+	// Example:
+	// Any endpoint with the label "role=backend" can be consumed by any
+	// endpoint carrying the label "role=frontend".
+	//
+	// +optional
+	FromEndpoints []EndpointSelector `json:"fromEndpoints,omitempty"`
+
+	// FromRequires is a list of additional constraints which must be met
+	// in order for the selected endpoints to be reachable. These
+	// additional constraints do no by itself grant access privileges and
+	// must always be accompanied with at least one matching FromEndpoints.
+	//
+	// Example:
+	// Any Endpoint with the label "team=A" requires consuming endpoint
+	// to also carry the label "team=A".
+	//
+	// +optional
+	FromRequires []EndpointSelector `json:"fromRequires,omitempty"`
+
+	// ToPorts is a list of destination ports identified by port number and
+	// protocol which the endpoint subject to the rule is allowed to
+	// receive connections on.
+	//
+	// Example:
+	// Any endpoint with the label "app=httpd" can only accept incoming
+	// connections on port 80/tcp.
+	//
+	// +optional
+	ToPorts []PortRule `json:"toPorts,omitempty"`
+
+	// FromCIDR is a list of IP blocks which the endpoint subject to the
+	// rule is allowed to receive connections from in addition to FromEndpoints.
+	// This will match on the source IP address of incoming connections.
+	//
+	// Example:
+	// Any endpoint with the label "app=my-legacy-pet" is allowed to receive
+	// connections from 10.3.9.1
+	//
+	// +optional
+	FromCIDR []CIDR `json:"fromCIDR,omitempty"`
```

Comment:
I've changed the CIDR struct to contain a string, that gives us the most flexibility going forward